sdk/logs/src/main/java/io/opentelemetry/sdk/logs/LoggerSharedState.java

```diff
@@ -50,6 +55,11 @@ Clock getClock() {
     return clock;
   }
 
+  LoggerConfig getLoggerConfig(InstrumentationScopeInfo instrumentationScopeInfo) {
```

Comment:
It looks like the getter here exists only so that the caller of this getter can ask the resulting `LoggerConfig` if `isEnabled()` (law of Demeter problem). What would you think about removing this getter and instead adding a method called `isConfigEnabled(scope)` instead? 

I think this is true of the other two signals as well.

Reply:
> It looks like the getter here exists only so that the caller of this getter can ask the resulting LoggerConfig if isEnabled() (law of Demeter problem). 

That's true for now, but I don't expect that to always be the case since I expect there to be other scope level config options, which may vary from signal to signal. The decision to keep the response be LoggerConfig was therefore forward looking: a logger asked the centralized shared state "hey, what's my resolved config? I know how to interpret that config to change behavior of logger"

You could make the argument that having `boolean isConfigEnabled(scope)` violates law of demeter in the other direction, since it requires LoggerSharedState to have knowledge about a specific aspect of logger config that it shouldn't need to know about.

But thinking about this more, logger config should be set when SdkLogger is initialized. And later, when we allow SDK config to be updated dynamically, some central point should respond to updated scope config and go update all the outstanding loggers with the new config. So SdkLogger will eventually need a `setLogger`. The central point which will update all the outstanding loggers will be the SdkLoggerProvider, which conveniently also is the thing that calls `new SdkLogger` today. So it seems like the more elegant solution is to:

- Add a private method to SdkLoggerProvider to compute the resolved scope config
- Call this method and pass the LoggerConfig when constructing new SdkLogger
- When we add the ability to dynamically update scope config, SdkLogger can iterate through all outstanding SdkLoggers, compute the new scope config, and call a new `SdkLogger#setLoggerConfig` method